tools/load_balancing_tool/load_balancing_submit.py

```diff
@@ -0,0 +1,326 @@
+          <rootpe_rof>0</rootpe_rof>
+          <rootpe_ice>0</rootpe_ice>
+          <rootpe_ocn>0</rootpe_ocn>
+          <rootpe_glc>0</rootpe_glc>
+          <rootpe_wav>0</rootpe_wav>
+          <rootpe_cpl>0</rootpe_cpl>
+        </rootpe>
+      </pes>
+    </mach>
+  </grid>
+
+  <grid name="any">
+    <mach name="any">
+      <pes compset="any" pesize="3">
+        <comment>none</comment>
+        <ntasks>
+          <ntasks_atm>128</ntasks_atm>
+          <ntasks_lnd>128</ntasks_lnd>
+          <ntasks_rof>128</ntasks_rof>
+          <ntasks_ice>128</ntasks_ice>
+          <ntasks_ocn>128</ntasks_ocn>
+          <ntasks_glc>128</ntasks_glc>
+          <ntasks_wav>128</ntasks_wav>
+          <ntasks_cpl>128</ntasks_cpl>
+        </ntasks>
+        <nthrds>
+          <nthrds_atm>1</nthrds_atm>
+          <nthrds_lnd>1</nthrds_lnd>
+          <nthrds_rof>1</nthrds_rof>
+          <nthrds_ice>1</nthrds_ice>
+          <nthrds_ocn>1</nthrds_ocn>
+          <nthrds_glc>1</nthrds_glc>
+          <nthrds_wav>1</nthrds_wav>
+          <nthrds_cpl>1</nthrds_cpl>
+        </nthrds>
+        <rootpe>
+          <rootpe_atm>0</rootpe_atm>
+          <rootpe_lnd>0</rootpe_lnd>
+          <rootpe_rof>0</rootpe_rof>
+          <rootpe_ice>0</rootpe_ice>
+          <rootpe_ocn>0</rootpe_ocn>
+          <rootpe_glc>0</rootpe_glc>
+          <rootpe_wav>0</rootpe_wav>
+          <rootpe_cpl>0</rootpe_cpl>
+        </rootpe>
+      </pes>
+    </mach>
+  </grid>
+</config_pes>
+"""
+    parser = argparse.ArgumentParser(usage=help_str,
+                                     description=description,
+                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    CIME.utils.setup_standard_logging_options(parser)
+    # Required arguments
+    parser.add_argument('--compset',
+                        help='Specify compset', required=True)
+    parser.add_argument('--res',
+                        help='Specify resolution', required=True)
+    parser.add_argument('--pesfile', required=True)
+
+    # Optional pass-through arguments to create_newcase
+    parser.add_argument('--compiler', help='Choose compiler to build with')
+    parser.add_argument('--project', help='Specify project id')
+    parser.add_argument('--machine', help='machine name')
+
+    parser.add_argument('--extra_options_file',
+                        help='file listing options to be run using xmlchange')
+    parser.add_argument('--casename_prefix', default=DEFAULT_CASENAME_PREFIX,
+                        help='casename prefix to use for all timing runs')
+    parser.add_argument('--force_purge', action='store_true')
+
+    args = CIME.utils.parse_args_and_handle_standard_logging_options(args, parser)
+
+    return (args.compset, args.res, args.pesfile,
+            args.compiler, args.project, args.machine, args.extra_options_file,
+            args.casename_prefix, args.force_purge)
+
+def _set_xml_val(option, value, casedir):
```

Comment:
It's not clear to me why you are using a shell interface to xmlchange instead of instantiating a case object and using case.set_value?